### DISCUSSION TITLE
Useful tools/scripts to consider developing?

### DISCUSSION THREAD

**Discussion by jukelennings** [pushsecurity/browser-identity-attacks-matrix#16] [Ideas]
Many potential tools for use in demonstrating or detecting attacks in this repository would end up being highly specific to both individual SaaS apps as well as a given attack - that's a huge problem to try to solve. However, it's possible there might be some areas where the main functionality is pretty common and anything specific to a given SaaS app would be lightweight and could be easily added in a plugin-like manner, so one tool could easily cover a lot of bases. The other option is tools that impact the top few largest SaaS apps like M365, Azure, Slack etc.

A couple of examples that might be good starting points that come to mind:

1) Tenant discover/enumeration methods - often these are via simple HTTP or DNS requests and so would be easy to add for lots of SaaS apps
2) Username enumeration methods - similarly, often simple HTTP requests
3) Credential stuffing - similarly, often simple HTTP requests

**Comment by jacques-** [comment]
@jukelennings you want to put each of those ideas in it's own comment so we can upvote specific ones?
